src/collections/queue/mod.rs

```diff
@@ -0,0 +1,80 @@
+/// A queue-like data structure implement through Vector
+pub struct Queue<T> {
+    items: Vec<T>,
+}
+
+impl<T> Queue<T> {
+    /// Initialize a queue with empty vector
+    pub fn new() -> Self {
+        Self {
+            items: Vec::new(),
+        }
+    }
+
+    /// Adds an element into queue
+    pub fn enqueue(&mut self, item: T) -> bool {
+        self.items.push(item);
+        true
+    }
+
+    /// Removes the oldest added element in queue
+    pub fn dequeue(&mut self) -> Option<T> {
+        match self.items.is_empty() {
+            false => Some(self.items.remove(0)),
+            true => None,
+        }
+    }
+
+    /// Retrieves the element that is the oldest added without dequeue
+    pub fn peek(&self) -> Option<&T> {
+        self.items.first()
+    }
+
+    /// Retrieves the size of the queue
+    pub fn size(&self) -> usize {
+        self.items.len()
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    #[test]
+    fn test_new() {
+        let queue = super::Queue::<i32>::new();
+        assert!(queue.items.is_empty());
+    }
+    #[test]
+    fn test_enqueue() {
+        let mut queue = super::Queue::<i32>::new();
+        queue.enqueue(32i32);
+        assert_eq!(Some(&32i32), queue.peek());
+        assert_eq!(1, queue.size());
```

Comment:
NIT: Just a styling issue🤪. Rust compiler is smart enough so we can remove explicit type argument and it can still inference the type correctly.

```suggestion
        let mut queue = super::Queue::new();
        queue.enqueue(32);
        assert_eq!(Some(&32), queue.peek());
        assert_eq!(1, queue.size());
```